### PR TITLE
[2.4] Backport AWS VPC/Subnet enhancement from main branch

### PR DESCRIPTION
**Pull request by IsaacYangSLA**
### Description

Backport this feature from main branch to 2.4.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/dashboard/cli.py [hunk 1]
```diff
@@ -191,6 +195,18 @@ def define_dashboard_parser(parser):
...
+        "--vpc-id",
+        type=str,
+        default="",
+        help="VPC id for AWS EC2 instance.  Applicable to AWS only.  Ignored if subnet-id is not specified.",
+    )
+    parser.add_argument(
+        "--subnet-id",
+        type=str,
+        default="",
+        help="Subnet id for AWS EC2 instance.  Applicable to AWS only.  Ignored if pvc-id is not specified.",
```
Comment (YuanTingHsieh) [inline nvflare/dashboard/cli.py:208]: do you mean pvc-id or vpc-id?

**Review by YuanTingHsieh — COMMENTED** [review]
one typo, other LGTM

File: nvflare/dashboard/cli.py (hunk 1 above)
Comment (IsaacYangSLA) [reply nvflare/dashboard/cli.py:208]: My bad.

**Pull request merged**